### PR TITLE
Add aws_codestarconnections_connection resource

### LINKED ISSUE #11389: Codepipeline Bitbucket Integration
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Codepipeline now supports integration with Bitbucket. Please refer this announcement
https://aws.amazon.com/about-aws/whats-new/2019/12/aws-codepipeline-now-supports-atlassian-bitbucket-cloud/

### New or Affected Resource(s)

* aws_codepipeline

### Potential Terraform Configuration

```hcl
resource "aws_codepipeline" "this" {
  name     = "${var.codecommit_name}-pipeline"
  role_arn = var.codepipeline_role_arn

  artifact_store {
    [..]
  }

  stage {
    name = "Source"

    action {
      name             = "Source"
      category         = "Source"
      owner            = "ThirdParty"
      provider         = "Bitbucket"
      version          = "1"
      output_artifacts = ["source_output"]

      configuration = {
        Owner  = "my-organization"
        Repo   = "test"
        Branch = "master"
      }
    }
    [..]
  }

  stage {
    name = "Build"

    action {
      name            = "BuildFrontend"
      category        = "Build"
      owner           = "AWS"
      provider        = "CodeBuild"
      input_artifacts = ["source"]

      output_artifacts = [
        [..]
      ]

      version = "1"

      configuration = {
        ProjectName = aws_codebuild_project.this.name
      }
    }

    action {
      name            = "BuildBackend"
      category        = "Build"
      owner           = "AWS"
      provider        = "CodeBuild"
      input_artifacts = ["source"]

      output_artifacts = [
        [..]
      ]

      version = "1"

      configuration = {
        ProjectName = aws_codebuild_project.this.name
      }
    }
  }

  stage {
    name = "DeployDevelopment"

    action {
      [..]
    }

    action {
      [..]
    }
  }
```

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.com/hashicorp/terraform-provider-aws/issues/15453, https://github.com/hashicorp/terraform-provider-aws/pull/15960
Closes #12637
Closes #11389

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* **New Resource:** `aws_codestarconnections_connection`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSCodeStarConnectionsConnection_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCodeStarConnectionsConnection_* -timeout 120m
=== RUN   TestAccAWSCodeStarConnectionsConnection_Basic
=== PAUSE TestAccAWSCodeStarConnectionsConnection_Basic
=== RUN   TestAccAWSCodeStarConnectionsConnection_disappears
=== PAUSE TestAccAWSCodeStarConnectionsConnection_disappears
=== CONT  TestAccAWSCodeStarConnectionsConnection_disappears
=== CONT  TestAccAWSCodeStarConnectionsConnection_Basic
--- PASS: TestAccAWSCodeStarConnectionsConnection_disappears (36.78s)
--- PASS: TestAccAWSCodeStarConnectionsConnection_Basic (49.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	51.265s
```

This is the second part of a PR to add `aws_codestarconnections_connection` resource (the first part: https://github.com/hashicorp/terraform-provider-aws/pull/15960). This PR originally came from https://github.com/hashicorp/terraform-provider-aws/pull/12637. Thank you for your review!

### CODE REVIEW

**Comment by shuheiktgw** [comment]
@bflad I've created the second part of the PR to support `aws_codestarconnections_connection` resource so would you review the PR?

File: aws/resource_aws_codestarconnections_connection.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+func resourceAwsCodeStarConnectionsConnectionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codestarconnectionsconn
+
+	_, err := conn.DeleteConnection(&codestarconnections.DeleteConnectionInput{
+		ConnectionArn: aws.String(d.Id()),
+	})
+
+	if err != nil {
```
Comment (gdavison) [inline aws/resource_aws_codestarconnections_connection.go:107]: This should also handle the case where the CodeStar connection is not found by returning immediately

File: aws/resource_aws_codestarconnections_connection_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
```
Comment (gdavison) [inline aws/resource_aws_codestarconnections_connection_test.go:43]: In order to test the ability of the provider to handle the resource being deleted outside of Terraform, this should include a "Disappears" test, as documented at https://github.com/hashicorp/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#disappears-acceptance-tests

**Review by gdavison — CHANGES_REQUESTED** [review]
Thanks for the PR, @shuheiktgw, it's looking good. I have a few suggestions for additions to the resource

**Comment by shuheiktgw** [comment]
@gdavison Thank you for your review! Let me fix them quickly 👍

**Comment by shuheiktgw** [comment]
@gdavison Thank you for your great reviews! I believe I fixed them up so would you review the PR again? 😄 
I also reran the acc tests so please check the results below.

```
$ make testacc TESTARGS='-run=TestAccAWSCodeStarConnectionsConnection_*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCodeStarConnectionsConnection_* -timeout 120m
=== RUN   TestAccAWSCodeStarConnectionsConnection_Basic
=== PAUSE TestAccAWSCodeStarConnectionsConnection_Basic
=== RUN   TestAccAWSCodeStarConnectionsConnection_disappears
=== PAUSE TestAccAWSCodeStarConnectionsConnection_disappears
=== CONT  TestAccAWSCodeStarConnectionsConnection_disappears
=== CONT  TestAccAWSCodeStarConnectionsConnection_Basic
--- PASS: TestAccAWSCodeStarConnectionsConnection_disappears (36.78s)
--- PASS: TestAccAWSCodeStarConnectionsConnection_Basic (49.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	51.265s
```

File: aws/resource_aws_codestarconnections_connection_test.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	conn := testAccProvider.Meta().(*AWSClient).codestarconnectionsconn
+
+	for _, rs := range s.RootModule().Resources {
+		switch rs.Type {
+		case "aws_codestarconnections_connection":
+			_, err := conn.GetConnection(&codestarconnections.GetConnectionInput{
+				ConnectionArn: aws.String(rs.Primary.ID),
+			})
+
+			if err != nil && !isAWSErr(err, codestarconnections.ErrCodeResourceNotFoundException, "") {
```
Comment (gdavison) [inline aws/resource_aws_codestarconnections_connection_test.go:97]: A note for future reference: `isAWSErr()` and the error checking functions at https://github.com/hashicorp/aws-sdk-go-base/blob/master/tfawserr/awserr.go can handle a `nil` error, so `err != nil` isn't technically needed

**Review by gdavison — APPROVED** [review]
Thanks for the updates, @shuheiktgw! LGTM 🚀 

--- PASS: TestAccAWSCodeStarConnectionsConnection_disappears (10.42s)
--- PASS: TestAccAWSCodeStarConnectionsConnection_Basic (14.08s)

**Pull request merged**